dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetFactoryIntegrationTest.java

```diff
@@ -0,0 +1,214 @@
+package com.dotmarketing.portlets.fileassets.business;
+
+import com.dotcms.IntegrationTestBase;
+import com.dotcms.datagen.*;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Permission;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.Permissionable;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.liferay.portal.model.User;
+import com.liferay.util.FileUtil;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class FileAssetFactoryIntegrationTest extends IntegrationTestBase {
+
+    private static FileAssetFactoryImpl fileAssetFactory;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        //Setting web app environment o
+        IntegrationTestInitService.getInstance().init();
+
+        fileAssetFactory = new FileAssetFactoryImpl();
+    }
+
+    /**
+     * Method to test: {@link FileAssetFactory#findFileAssetsByFolderInDB(Folder, User, boolean)}
+     * When: Create two Folder with two files each one, and try to get the files from the first folder
+     *       and the user jave permission over all the files
+     * Should: return only the two files into the first folder
+     *
+     * @throws IOException
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void shouldReturnTwoFiles() throws IOException, DotDataException, DotSecurityException {
+        final Role backEndUserRole = APILocator.getRoleAPI().loadBackEndUserRole();
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role, backEndUserRole).nextPersisted();
+
+        final Host host = new SiteDataGen().nextPersisted();
+
+        final Folder folder1 = new FolderDataGen().site(host).nextPersisted();
+        final Contentlet fileAsset1 = createFileAsset(folder1, "text1", ".txt");
+        final Contentlet fileAsset2 = createFileAsset(folder1, "text2", ".txt");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "text2" appears 4 times in this file; the first occurrence is on line 60](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280157)